### PR TITLE
Add Study Lesson Length

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add Rails config study length to display lessons.
* Add defaults if host apps don't have necessary configurations.
* Remove rescue and default statements.

[#104761294]

### CODE REVIEW

File: config/initializers/think_feel_do_engine.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Rails.application.configure do
+  config.lesson_week_length = 0
+  config.study_length_in_weeks = 0
```
Comment (ericcf) [inline config/initializers/think_feel_do_engine.rb]: When you set this in a host app initializer, I assume it overrides it?

Comment (michaelwehrley) [inline config/initializers/think_feel_do_engine.rb]: yep. :smile:

File: spec/features/participant/learn/content_providers/learn_lesson_provider_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        before do
+          Rails.application.config.lesson_week_length = 0
+        end
+
+        after do
+          Rails.application.config.lesson_week_length = 16
+        end
+
+        scenario "No lessons are available for any weeks" do
```
Comment (michaelwehrley) [inline spec/features/participant/learn/content_providers/learn_lesson_provider_spec.rb]: @ericcf scenario description update made.

**Comment by ericcf** [comment]
@michaelwehrley I'm fine with this. Please just squash the commits and merge. Thanks!

**Comment by michaelwehrley** [comment]
How did that get past me :-)

**Pull request merged**